sdk/ml/azure-ai-ml/azure/ai/ml/operations/_evaluator_operations.py

```diff
@@ -0,0 +1,772 @@
+        :type label: str
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/ml_samples_misc.py
+                :start-after: [START model_operations_restore]
+                :end-before: [END model_operations_restore]
+                :language: python
+                :dedent: 8
+                :caption: Restore an archived model.
+        """
+        self.get(name=name, version=version, label=label)
+        _archive_or_restore(
+            asset_operations=self,
+            version_operation=self._model_versions_operation,
+            container_operation=self._model_container_operation,
+            is_archived=False,
+            name=name,
+            version=version,
+            label=label,
+        )
+
+    @monitor_with_activity(ops_logger, "Evaluator.List", ActivityType.PUBLICAPI)
+    def list(
+            self,
+            name: str,
+            stage: Optional[str] = None,
+            *,
+            list_view_type: ListViewType = ListViewType.ACTIVE_ONLY,
+    ) -> Iterable[Model]:
+        """List all model assets in workspace.
+
+        :param name: Name of the model.
+        :type name: Optional[str]
+        :param stage: The Model stage
+        :type stage: Optional[str]
+        :keyword list_view_type: View type for including/excluding (for example) archived models.
+            Defaults to :attr:`ListViewType.ACTIVE_ONLY`.
+        :paramtype list_view_type: ListViewType
+        :return: An iterator like instance of Model objects
+        :rtype: ~azure.core.paging.ItemPaged[~azure.ai.ml.entities.Model]
+        """
+        if name:
+            return cast(
+                Iterable[Model],
+                (
+                    self._model_versions_operation.list(
+                        name=name,
+                        registry_name=self._registry_name,
+                        cls=lambda objs: [Model._from_rest_object(obj) for obj in objs],
+                        **self._scope_kwargs,
+                    )
+                    if self._registry_name
+                    else self._model_versions_operation.list(
+                        name=name,
+                        workspace_name=self._workspace_name,
+                        cls=lambda objs: [Model._from_rest_object(obj) for obj in objs],
+                        list_view_type=list_view_type,
+                        properties="is-promptflow=true,is-evaluator=true",
+                        # tags={"type": "evaluator"},
+                        stage=stage,
+                        **self._scope_kwargs,
+                    )
+                ),
+            )
+
+        return cast(
+            Iterable[Model],
+            (
+                self._model_container_operation.list(
+                    registry_name=self._registry_name,
+                    cls=lambda objs: [Model._from_container_rest_object(obj) for obj in objs],
```

Comment:
Is it possible to do client side filter here?

Reply:
Fixed, however, we will need to add filtering for model container operations, so that it will be able to handle properties.